src/serde/ser/struct_serializer.rs

```diff
@@ -0,0 +1,48 @@
+use crate::serde::common::*;
+
+use serde::ser::SerializeStruct;
+
+/// Bencode sub-serializer for structs.
+pub struct StructSerializer<'outer> {
```

Comment:
This code is largely duplicated from UnsortedDictEncoder. Would it be possible to implement this in terms of UnsortedDictEncoder, or did you run into a problem trying to do so?

Reply:
I updated the code to reuse as much as possible from the `UnsortedDictEncoder`. I still needed a new struct though, since it needs access to the outer `Serializer` object when the struct is finished.